pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -238,6 +238,16 @@ func newOpenStack(cfg Config) (*OpenStack, error) {
 }
 
 func getServerByName(client *gophercloud.ServiceClient, name string) (*servers.Server, error) {
+	// try to read from local file (cloud-init) or metadata server first
+	id, err := readInstanceID()
+	if err == nil {
+		server, err := servers.Get(client, id).Extract()
+		if err == nil {
+			return server, nil
```

Comment:
I'm not sure at what level this would happen.  If the nodeName is not required to map in any way to the Openstack instance name/id, how would you detect that the name you are looking for is the local node?

Also, the rackspace code (link in the first comment) already assumes this, so if the assumption breaks, it will break for at least them as well.  I have confirmed in the code this is not called from outside the kubelet.

I do agree that ignoring the name in a function called ```getServerByName()``` is weird